doh-server/config.go

```diff
@@ -57,6 +58,10 @@ func loadConfig(path string) (*config, error) {
 		conf.Listen = []string{"127.0.0.1:8053", "[::1]:8053"}
 	}
 
+	if conf.LocalAddr == "" {
+		conf.LocalAddr = conf.Listen[0]
```

Comment:
Would you prefer to keep the existing behavior when `conf.LocalAddr == ""`, i.e. letting [`dns.Client`](https://github.com/miekg/dns/blob/master/client.go#L29-L44) instantiate its own `Dialer` with the default local address?

Reply:
I went ahead and did that. Let me know what you think.